flamingpy/utils/viz.py

```diff
@@ -11,64 +11,109 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Helper functions to draw various graphs and generate plots using
-Matplotlib."""
+"""Helper functions to draw various graphs and generate plots using Matplotlib.
+
+Plots are configured via the `plot_params` dictionary. This parameters
+are associated with Matplolib's rc settings and are modified within the
+plotting functions using `rc_context` context manager. This approach
+avoids having to modify the global Matplotlib `rc_params`.
+
+To modify the plot parameters use, for example,
+
+  .. code::
+
+    from flamingpy.utils.viz import plot_params as fp_plot_params
+    fp_plot_params["font.size"] = 20
+"""
 
 # pylint: disable=too-many-statements
 
 import itertools as it
 
 import numpy as np
 import networkx as nx
+import matplotlib as mpl
 from matplotlib.patches import Patch
 import matplotlib.pyplot as plt
 import matplotlib.lines as mlines
 
 from flamingpy.codes import Stabilizer
 from flamingpy.cv import gkp
 
-
+plot_params = {
+    "font.size": 12,
+    "font.family": "sans-serif",
+    "axes.labelsize": 13,
+    "axes.titlesize": 17,
+    "xtick.labelsize": 13,
+    "ytick.labelsize": 13,
+    "legend.fontsize": 13,
+    "grid.color": "lightgray",
+    "lines.markersize": 5,
+    "lines.linewidth": 5,
+    "figure.figsize": (8, 6),
+}
+
+
+@mpl.rc_context(plot_params)
 def plot_integer_part(xs, ns, alpha, show=True):
     """Plot the integer part of real numbers mod alpha."""
     xmin, xmax = alpha * (xs[0] // alpha), alpha * (xs[-1] // alpha) + alpha
     newxticks = np.linspace(xmin, xmax, int((xmax - xmin) // alpha) + 1)
     newxlabels = [gkp.to_pi_string(tick) for tick in newxticks]
-    plt.plot(xs, ns, ",")
-    plt.title("Integer Part", fontsize="medium")
-    plt.xticks(newxticks, newxlabels, fontsize="small")
+    fig = plt.figure()
+    ax = plt.gca()
+    plt.plot(xs, ns, ".")
+    plt.title("Integer Part")
+    plt.xlabel("$x$")
+    plt.xticks(newxticks, newxlabels)
+    plt.ylabel(r"$\mathrm{int}(x)$")
     if show:
         plt.show()
 
+    return fig, ax
+
 
+@mpl.rc_context(plot_params)
 def plot_fractional_part(xs, fs, alpha, show=True):
     """Plot the fractional part of real numbers mod alpha."""
-    plt.title("Fractional Part", fontsize="medium")
-    plt.plot(xs, fs, ",")
+    plt.title("Fractional Part")
+    plt.plot(xs, fs, ".", linewidth=10)
```

Comment:
```suggestion
    plt.plot(xs, fs, ".")
```
Not sure we need linewidth since this is set by plot params, and we have point markers instead of lines here anyway